activestorage/app/models/active_storage/attachment.rb

```diff
@@ -131,20 +131,23 @@ def mirror_blob_later
       blob.mirror_later
     end
 
-    def transform_variants_later
-      preprocessed_variations = named_variants.filter_map { |_name, named_variant|
-        if named_variant.preprocessed?(record)
-          named_variant.transformations
-        end
-      }
+    def create_variants
```

Comment:
Maybe rename this to make clear variants might not get created immediately?
```suggestion
    def schedule_variant_transformations
```

Reply:
> Maybe rename this to make clear variants might not get created immediately?

Yeah, really it would be `create_or_schedule_create_variants`. My thinking was the method `create_variants` creates the variants based on their own creation strategy. 🤷‍♂️